etcdctl/command/cluster_health.go

```diff
@@ -46,9 +46,10 @@ func handleClusterHealth(c *cli.Context) {
 	}
 
 	// do we have a leader?
-	ep, ls0, err := getLeaderStats(tr, client.GetCluster())
+	cl := client.GetCluster()
+	ep, ls0, err := getLeaderStats(tr, cl)
 	if err != nil {
-		fmt.Println("cluster is unhealthy")
+		fmt.Println("failed to get leader stats from cluster urls", cl)
```

Comment:
I am not sure if this can help... 

In fact, in this case they cluster might be unhealthy indeed. that is the reason you cannot contact the leader. 

Then this error message is misleading, since user might think the cluster is OK, the problem is the client by looking at `failed to get leader stats from cluster urls`.

There is no way to determine if a distributed system is OK from a 3rd party actually.